addons/mail/models/mail_thread.py

```diff
@@ -3132,6 +3157,24 @@ def _notify_thread(self, message, msg_vals=False, **kwargs):
 
         return recipients_data
 
+    def _notify_thread_by_channel(self, messages, **kwargs):
+        """ Send bus notifications for messages through channels.
+
+        :param recordset messages: A recordset of mail.message records being
+         notified. It may be void as 'msg_vals' supersedes it.
+        """
+        bus_notifications = []
+        if messages:
+            for message in messages:
+                message_format = message._message_format()[0]
```

Comment:
```suggestion
            for message_format in message._message_format():
```

And below where you access `message['res_id']` could be replaced by `message_format['res_id']`.